### DISCUSSION TITLE
Related work

### DISCUSSION THREAD

**Discussion by dabreegster** [a-b-street/osm2streets#195] [General]
This is an issue just to keep track of other related work.

https://gisintransportation.com/media/zsnn3oou/2022_04_19-gis-t-workshop-aegist-slides.pdf, see page 18. They have a nice classification of some complex junctions:
![Screenshot from 2022-06-29 18-51-01](https://user-images.githubusercontent.com/1664407/176564551-b8711fc4-421c-4b9e-83c5-c47cfc99e2c2.png)

**Comment by dabreegster** [comment]
Apparently sharedstreets does some simplifications of small roads, but doesn't collapse dual carriageways. The most detail I've found is in the code: https://github.com/sharedstreets/sharedstreets-builder/blob/a554983e96010d32b71d7d23504fa88c6fbbad10/src/main/java/io/sharedstreets/tools/builder/transforms/BaseSegments.java#L210

**Comment by dabreegster** [comment]
https://github.com/gsagostini/momepy/blob/main/gsocc/notebooks/22-06-30_intersections.ipynb (ongoing google summer of code project)

**Comment by dabreegster** [comment]
https://wiki.openstreetmap.org/wiki/Berlin/Verkehrswende/Gehwege has been updated since I last looked -- there are some awesome diagrams there

**Comment by dabreegster** [comment]
https://sdna-open.readthedocs.io/en/latest/step_by_step_guides.html#modelling-a-combined-vehicle-and-cycle-network found by @fhk

**Comment by dabreegster** [comment]
https://hal.archives-ouvertes.fr/hal-02280105/document averages parallel rail lines

**Comment by dabreegster** [comment]
https://traffic3d.org/import_open_street_map.html (it's a new traffic simulator!)

**Comment by fhk** [comment]
Also saw this - https://muffinman.io/blog/draw-svg-rope-using-javascript/

Interesting write up and geometric ideas

**Comment by dabreegster** [comment]
Someone sent along https://peertube.openstreetmap.fr/w/kF7FsouSL6UwafhwTUgPYQ?start=17m48s

**Comment by dabreegster** [comment]
https://community.openstreetmap.org/t/tagging-a-sidewalk-name-or-creating-relation-to-street/7786/23

**Comment by BudgieInWA** [comment]
I haven't read this yet, but saw a recent paper tackling this problem "From road centrelines to carriageways—A
reconstruction algorithm": https://journals.plos.org/plosone/article/file?id=10.1371/journal.pone.0262801&type=printable

**Comment by BudgieInWA** [comment]
"OSM-Based Automatic Road Network Geometries Generation on Unity" https://kth.diva-portal.org/smash/get/diva2:1375175/FULLTEXT01.pdf

**Comment by dabreegster** [comment]
Super cool finds! I skimmed through both quickly, but need to read closely later to understand. The first one's repo is https://github.com/tudelft3d/carriageways-creator, but I'm having trouble building it

**Comment by fhk** [comment]
Nice! find @BudgieInWA

**Comment by dabreegster** [comment]
https://github.com/kauevestena/osm_sidewalkreator
The slide deck from SoTM is very helpful -- this is feeling like an intersection geometry algorithm:
![Screenshot from 2023-03-08 17-16-39](https://user-images.githubusercontent.com/1664407/223783503-16fc7694-333a-44fa-819b-7e1805bb8313.png)

**Reply by kauevestena** [reply]
Hey, hello! I am so glad to see people noticing my plugin =D

**Reply by kauevestena** [reply]
But this design has its flaws; I'm working on a new algorithm for such a problem!

**Comment by dabreegster** [comment]
https://github.com/anisotropi4/graph/blob/master/graph.md possibly relates to collapsing parallel ways

**Comment by dabreegster** [comment]
https://stamen.com/merging-offline-edits-with-the-posm-replay-tool-2f39a4410d2a/#.hi4xe5hlw is about resolving diffs with upstream OSM data

**Comment by dabreegster** [comment]
Streets GL is awesome! https://streets.gl/#47.60185,-122.33009,81.00,359.25,407.24, https://github.com/strandedkitty/streets-gl.
![Screenshot from 2023-05-04 16-40-39](https://user-images.githubusercontent.com/1664407/236258488-05bdc877-09b9-4eea-9c49-6ebc84672d6e.png)
![Screenshot from 2023-05-04 16-40-02](https://user-images.githubusercontent.com/1664407/236258496-7ff73a4f-30db-4792-af80-7ef8807b2210.png)

Intersection geometry: https://github.com/StrandedKitty/streets-gl/blob/dev/src/lib/road-graph/IntersectionPolygonBuilder.ts
Road parsing: https://github.com/StrandedKitty/streets-gl/blob/dev/src/lib/tile-processing/vector/qualifiers/factories/PolylineQualifierFactory.ts

**Comment by dabreegster** [comment]
[Overture Maps](https://github.com/OvertureMaps/data/blob/main/README.md) just released data. They have a JSON schema for lanes(https://docs.overturemaps.org/reference/transportation/segment) and, presumably, parsing from OSM tags. An example at https://github.com/OvertureMaps/schema/blob/main/examples/transportation/segment/road/road-oneway-with-lanes.yaml. Anybody explore this data yet?

**Reply by fhk** [reply]
wondering if they will add width and perhaps height limit?

**Reply by BudgieInWA** [reply]
I saw the post but haven't dived in yet. Looks like a great opportunity to "standardise" on the good ideas.

**Comment by BudgieInWA** [comment]
I think the [GERS IDs](https://docs.overturemaps.org/gers/) and [Linear Referencing](https://docs.overturemaps.org/themes/transportation/#geometric-scoping-linear-referencing) is super interesting too. [OpenLR](http://www.openlr.org/method.html) does linear referencing already.

**Comment by dabreegster** [comment]
Mapbox just released a new "Standard" style that has zebra crossings: https://labs.mapbox.com/labs-standard/#18.46/51.4953464/-0.1019343/0/18
No real lane detail or road width/shape, so pretty small in the context here

**Comment by fhk** [comment]
has some math for street width and geom - https://ths.rwth-aachen.de/wp-content/uploads/sites/4/schulz20203d.pdf

**Comment by dabreegster** [comment]
Network simplification: https://www.youtube.com/watch?v=F8yPKpUfQfU, https://github.com/achic19/micro_walking/blob/master/notebook/output/turin/data_from_osm.ipynb
@Robinlovelace, maybe useful for your current work

**Reply by Robinlovelace** [reply]
Awesome, thank you Dustin!

**Comment by tordans** [comment]
https://youtu.be/V_L-CaPWk1Y?si=-WhmLp5w-Qnp96FE demo of a OSM Import to create a street network for a game with lanes and such.

<img width="934" alt="image" src="https://github.com/a-b-street/osm2streets/assets/111561/c9bef227-9ed6-4658-8c52-df0dfc03b3d0">

**Comment by dabreegster** [comment]
https://ebikecity.ch/en.htm
Scroll down to their map!
![image](https://github.com/user-attachments/assets/60309216-d6ea-427e-a153-cf2ee759dbab)
They're just offsetting and thickening road center lines, but it still looks nice

**Comment by dabreegster** [comment]
https://doi.org/10.1016/j.jcmr.2024.100048 section 4.4, https://github.com/lukasballo/snman for simplifying parallel roads

**Comment by tordans** [comment]
**SOTM 2024: From Complexity to Clarity: Simplifying OpenStreetMap Data for Improved Active Transportation**

- https://youtu.be/fW82AvpyXW0 
- https://2024.stateofthemap.org/sessions/LAZXYU/
- Presentation https://pretalx.com/media/state-of-the-map-2024-academic-track/submissions/LAZXYU/resources/Final_presentaion__u2CXm6N.pdf
- It looks like the code is in those notebooks at https://github.com/achic19/SOD/tree/master/Code/notebooks

<details><summary>Talk Abstract</summary>


Achituv Cohen
OpenStreetMap (OSM) provides detailed street networks essential for analyzing active transportation (AT) infrastructure. However, the granularity and inconsistencies in OSM data pose challenges in modeling AT users’ movements at the street level. This study proposes a novel methodology to generate axial networks for AT users using OSM data, simplifying the network while preserving topology. Applied to cities like Turin, Tel Aviv, and San Francisco, this approach effectively streamlined pedestrian and bicycle networks. The proposed methodology can enhance AT infrastructure, contributing to safer and more efficient urban mobility.

Introduction OpenStreetMap (OSM) offers comprehensive street networks that span nearly every city worldwide. The networks contain essential details like road type and name, and individual streets are often represented through multiple concurrent segments. The segments include separate lanes for motor vehicles, dedicated bike paths, and pedestrian walkways. OSM’s comprehensive dataset allows the creation of specialized networks customized for pedestrian and cyclist analysis, providing users with a powerful tool for understanding and improving active transportation (AT) infrastructure. These specialized networks play a pivotal role in monitoring and understanding walking and bicycling patterns, contributing to infrastructure enhancement aligned with cities’ goals for fostering AT to combat traffic congestion, obesity, and air pollution (Nelson et al., 2021) Nevertheless, the granularity of these networks may pose challenges in modeling AT users’ movements on a street level. For example, when assessing AT safety or suggesting a walkability index at the street level, the current data representativeness on OSM necessitates significant manipulation to do so. Furthermore, owing to OSM’s open-editing model, the standards for mapping elements are not consistently defined, leading to contentions and variations in data quality (Haklay, 2010). Individuals often map elements based on personal needs and knowledge, introducing inconsistencies in the dataset. Consequently, in some locales, all designated lanes for various users are meticulously mapped, while in others, only a single lane representing the presence of a street is depicted. Additionally, there are instances where only lanes for motor vehicles are detailed, with scant attention paid to lanes catering to other road user groups. Previous research has endeavored to address this issue. For example, a study suggested a topology-preserving simplification of OSM network data for large-scale simulation in sumo (Meng et al., 2022). However, they primarily focus on less complex areas and prioritizing vehicular considerations when generating the new network over those of AT users. Aim In this study, we propose an innovative solution to generate an axial network specifically designed for monitoring and analyzing AT users, using exclusively OSM data. Our approach simplifies the network while preserving its topology. We apply this solution across diverse spatial contexts, from straightforward geographic regions to complex urban environments. Furthermore, we implement our methodology in several cities worldwide—Tel Aviv (Israel), Turin (Italy), and San Francisco (United State)—each characterized by unique urban structures and varying levels of economic development.

Methodology The preliminary tasks use OSMnx (Boeing, 2017) to acquire OSM street network data, converting it into a graph while correcting topological errors. Then, the data is stored in a geodata table, including polyline geometry, names, and road types. Our algorithm filters out unsuitable roads, like motorways and trunk roads, and replaces roundabouts with their central points. The network is then ready for the multilane detection algorithm. Polylines identified in a multilane scenario are aggregated into a centerline and added to the Simplification OSM Data (SOD) network. Other polylines are added to the SOD network, retaining their original geometry. A multilane scenario is identified when polylines share the same street name, have similar angles, and are close together. Polylines are grouped by street name, and azimuth (0°-180°) narrows the angle range to ensure that parallel lines are considered parallel, regardless of orientation. The Density-Based Spatial Clustering of Applications with Noise (DBSCAN) clusters similarly angled polylines using a 10° radius and a minimum of two samples. Outliers with significantly different angles are excluded. The remaining clusters apply a right-shifted buffer to each polyline. If two or more polylines in the same class overlap by at least 10% in the shifted buffer, they’re classified as multilane, and the entire class is replaced with one or more centerlines. The core idea behind creating a new centerline is to identify its start and end vertices and then add intermediate vertices to preserve the overall shape of multilane polylines. Overlapping buffers are merged into a single polygon, and the two polygon vertices that best define the original polylines are used to establish the new centerline’s start and end. Intermediate vertices are then added at regular intervals to maintain the original polylines’ orientation. The simplification process introduces significant changes to the locations of many polylines, resulting in issues of continuity, topology, accuracy, and inconsistency in the SOD network. Most problems can be resolved using existing methods, but connecting roundabouts requires extra effort. This process has three steps: transforming the roundabout representation to a point, connecting nearby dead-end polylines, and linking polylines in proximity. Following these steps ensures proper continuity and maintains the network’s topology.

Findings The case study centered on Turin, Italy, simplified its street network to 11,807 polylines from an initial 49,750. This effort reduced the complexity of 411 streets, converting 138 roundabouts to single points. Despite some challenges near intersections, the final version preserved topology. For validation, 43 streets in Turin were reviewed, revealing a 51% success rate, 28% with minor flaws, and 14% with partial success. Issues such as threshold errors, external mapping inaccuracies, and ambiguous street configurations affected the results. Similar challenges were faced in Tel Aviv, Israel, where 50% of 22 test streets matched perfectly with the reference network. Yet, 23% had minor issues, and 9% were entirely incorrect. In San Francisco, the comparison showed 80% accuracy with 20 streets evaluated. While most streets were correctly simplified, minor flaws were present in a few cases, mainly due to inconsistencies between the SOD and reference networks. Overall, the methodology successfully streamlined street networks, although it achieved varying rates of success across different cities.

Discussion Our study offers a robust methodology for generating axial networks for AT users using OSM data, balancing data simplification with topology preservation. Its successful application across diverse cities such as Turin, Tel Aviv, and San Francisco demonstrate its versatility and effectiveness in varied urban environments. The approach simplifies the complex urban network data, streamlining pedestrian and bicycle analysis while retaining essential details. This enables urban planners and policymakers to better monitor and understand AT patterns, leading to infrastructure improvements that support safer, more efficient, and environmentally friendly urban mobility. Our research offers both scientific contributions and practical benefits. Scientifically, it has been applied to evaluate the built environment for pedestrian walkability using a spatial data clustering approach. Additionally, it has been utilized in research to evaluate safety and bike network connectivity, aiming to improve bicycle and pedestrian infrastructure in California cities. Practically, this work has been published on GitHub, making the code accessible to everyone for their specific needs and goals.

References Boeing, G. (2017). OSMnx: New methods for acquiring, constructing, analyzing, and visualizing complex street networks. Computers, Environment and Urban Systems, 65, 126–139. Haklay, M. (2010). How good is volunteered geographical information? A comparative study of OpenStreetMap and Ordnance Survey datasets. Environment and Planning B: Planning and Design, 37(4), 682–703. Meng, Z., Du, X., Sottovia, P., Foroni, D., Axenie, C., Wieder, A., Eckhoff, D., Bortoli, S., Knoll, A., & Sommer, C. (2022). Topology-Preserving Simplification of OpenStreetMap Network Data for Large-scale Simulation in SUMO. SUMO Conference Proceedings, 3, 181–197. Nelson, T., Ferster, C., Laberee, K., Fuller, D., & Winters, M. (2021). Crowdsourced data for bicycling research and practice. Transport Reviews, 41(1), 97–114.



</details>

**Comment by dabreegster** [comment]
https://codepen.io/almccon/pen/KwKwWaz from https://osmus.slack.com/archives/C01G3D28DAB/p1739430672401849

**Comment by tordans** [comment]
There is a recent discussion at https://community.openstreetmap.org/t/quick-poll-lane-count/126298 and other channels around how the `lanes=<number>` tag should be interpreted which showed, that we cannot rely on that tag as much as we thought we could. I thought I post it here because all the projects here will somehow rely on that tag…

**Comment by tordans** [comment]
**Simplification of street networks** by @martinfleis
- https://martinfleischmann.net/simplification-of-street-networks/
- https://fosstodon.org/@martinfleis/114417530407685560
- https://github.com/uscuni/neatnet
- https://uscuni.org/neatnet/
- Paper https://arxiv.org/abs/2504.16198
- Slides https://uscuni.org/talks/slides/202504_GISRUK_simplification.html
   - A comparison of existing solutions https://uscuni.org/talks/slides/202504_GISRUK_simplification.html#/5/2

**Reply by tordans** [reply]
I first thought that was related to https://github.com/a-b-street/osm2streets/discussions/195#discussioncomment-11402241 but it is something new / different

**Comment by tordans** [comment]
> A global urban road network self-adaptive simplification workflow from traffic to spatial representation
> https://www.nature.com/articles/s41597-025-05164-9

According to https://datasci.social/@mszll/114606288002466513 this looks similar to https://github.com/a-b-street/osm2streets/discussions/195#discussioncomment-13167446

**Comment by tordans** [comment]
For reference, I wanted to write down what I am looking for in our work with OSM and simplified road networks. Mainly, because it looks like many of the solutions look at only one part of this, yet:

I am looking for a way to get the data into OSM processing that will allow me to know which ways are part of the same road space.
- when I process way 1, I want to know that this is a side path to road 2 and that road 2 is of type "primary"
- when I visualize roads, I want to draw a lane-line line that shows way 1 next to way 2, followed by way 3

This required information about the relation of roads in their shared road space. But also about their attributes.

**Reply by tordans** [reply]
Example 1: The CQI would ideally have one geometry that represents all lane-like structure (centerline, cycleway and or footway left left and right). Right now, we have messy situations like https://www.osm-verkehrswende.org/cqi/map/?anzeige=cqi&map=18.2%2F13.45162%2F52.47809&filters=usable-yes where tagging and mapping practices result in very different visualizations that don't help the presentation.

**Reply by tordans** [reply]
Example 2: Many external data sources use simplified road geometries (Edges in a Graph model). This make matching OSM Data very complex. We need to match the attributes but still keep data about the OSM ways and the relation to OSM. [This is an example](https://tilda-geo.de/regionen/berlin?map=16.8/52.5166/13.3918&config=1swjsz2.5ovxms.4qfsxw.pbd&v=2&data=radverkehrsnetz) where the red+orange line are external data and blue+yellow is OSM.

**Reply by dabreegster** [reply]
Your second example might be best-suited for something like map-matching. https://github.com/JosiahParry/anime is one approach.

I absolutely agree about preserving all of the semantics about the separate pieces of roads in the final simplified representation. I wrote up https://github.com/dabreegster/road-bundler with what I'm working on right now. If you want to play with it so far:

https://dabreegster.github.io/road-bundler/#16.02/52.478075/13.44795
Click the sidepath tool, then "Remove all sidepaths". Ignore the "merge" tool; I don't think this approach is the one that makes sense. Then go to the dual carriageway tool, click a purpleish area to simplify it. If you hover on the orange simplified edge, you see the original OSM edges that got "zipped" into it:
![image](https://github.com/user-attachments/assets/27807d78-7600-4781-82f4-e45af6ee01b0)
My goal is to put semantic data on that orange edge like "from left-to-right, there's a sidewalk (way 123, with tags XYZ), then the northbound road (way 456, tags ...), then southbound road, then another sidewalk".

This is still an early experiment, there are lots and lots of bugs.

**Comment by tordans** [comment]
https://www-openstreetmap-org.translate.goog/user/Mikhail%20Kuzin/diary/406938?_x_tr_sl=auto&_x_tr_tl=de&_x_tr_hl=de&_x_tr_pto=wapp
([Non-translated](https://www.openstreetmap.org/user/Mikhail%20Kuzin/diary/406938))

> ![image](https://github.com/user-attachments/assets/0e8d734c-a269-4146-a4fa-8d5edcc5a8e3)

**Reply by dabreegster** [reply]
Oh wow, this looks incredible. I couldn't find a Github repo anywhere or a link to the tool; maybe I missed it in the translation?

**Reply by kuzinmv** [reply]
Hello! Nice to meet you.
This app is not released yet, we plan to publish the app this fall, God willing. There is still a lot of work to do on the editor, the engine is more or less stable.

I am publishing chapters in the diary for now to evaluate the community's reaction.

Other draft articles will soon appear in the diaries. No translation yet, but by the official release they will be in English.

I will be happy to answer questions.

**Reply by dabreegster** [reply]
I look forward to the release then! It's a hugely difficult technical problem,and I'm very impressed from your first post. An app like this can help people fix OSM lane tagging so much faster than anything else, and demonstrate the power of detailed tagging. Let us know if you want any help with testing or feedback. https://osm2streets.org/ has a list of example areas that have proven to be difficult in this project, if you're looking for test cases.

**Reply by kuzinmv** [reply]
Thank you very much.
To make the wait less tedious, I published several articles in the diary.
https://www.openstreetmap.org/user/Mikhail%20Kuzin/diary

It would be interesting to know your opinion on the proposals for new tags. There is still time to fix something.

**Reply by kuzinmv** [reply]
> I look forward to the release then! It's a hugely difficult technical problem,and I'm very impressed from your first post. An app like this can help people fix OSM lane tagging so much faster than anything else, and demonstrate the power of detailed tagging. Let us know if you want any help with testing or feedback. https://osm2streets.org/ has a list of example areas that have proven to be difficult in this project, if you're looking for test cases.



Thank you for your feedback, and I want to please you a little. We are working on the official launch of OSMPIE

The application itself is now available online, as well as some [documentation](https://github.com/kuzinmv/osmpie-doc/blob/master/en/index.md), descriptions and [cool examples](https://github.com/kuzinmv/osmpie-doc/blob/master/en/examples/examples.md).

https://osmpie.org/

I’m also actively trying to introduce corrections to the [OSM roads](https://www.openstreetmap.org/user/Mikhail%20Kuzin/history#map=13/54.96684/73.40002) near [to me](https://osmpie.org/app/editor?pos=73.372939&pos=54.982285&zoom=15.22&tile=Carto+Light&bakeId=6952d032-a56c-438f-ba81-6301bc8e1359) . There you can see links to changesets to compare before and after and see which tags were used where.

There are still a lot of bugs in the editor and engine, but it is already quite usable.

That is why we suggest you try it now and of course we are waiting for your comments and [issues on GitHub](https://github.com/kuzinmv/osmpie-doc/issues).

**Comment by Robinlovelace** [comment]
Map visualising how Zurich would look after reallocating 50% space for active travel:

<img width="1912" height="784" alt="image" src="https://github.com/user-attachments/assets/0f6cf054-3f68-4633-b51f-2cb570c680bb" />

Source: https://www.ebikecity.ch/en.htm

**Reply by Robinlovelace** [reply]
In conversation with @hussein-mahfouz

**Reply by Robinlovelace** [reply]
I now realise this is a duplicate, apologies! https://github.com/a-b-street/osm2streets/discussions/195#discussioncomment-11283134

**Comment by Robinlovelace** [comment]
Just came across this open access paper on methods for extracting road space usage info from 3D point cloud LiDAR data: https://www.sciencedirect.com/science/article/pii/S1569843225004509

It demonstrates methods for classifying space at different levels of granularity with reference to OSM data

<img width="664" height="431" alt="image" src="https://github.com/user-attachments/assets/8a3064e5-eb3a-424a-b4b7-43fe9d020e8d" />

Thanks @valenca13 and heads-up @wangzhao0217 as related to the work you did on road width estimation.

**Comment by dabreegster** [comment]
From https://blog.opencagedata.com/post/openstreetmap-interview-osmpie-mikhail-kuzin, some webmaps with both a high level of road detail and great UX for selecting POIs / buildings:

https://yandex.com/maps/66/omsk/?indoorLevel=1&ll=73.372979%2C54.993904&mode=poi&poi%5Bpoint%5D=73.367743%2C54.994473&poi%5Buri%5D=ymapsbm1%3A%2F%2Forg%3Foid%3D208868946265&z=19.01

https://2gis.ru/omsk/geo/282213711094197/73.382814%2C54.970556?m=73.379572%2C54.972465%2F18.5&immersive=on

**Reply by tordans** [reply]
Also https://community.openstreetmap.org/t/complaint-and-criticism-against-osmpie-for-redefining-and-promoting-poorly-thought-tags-without-discussing-and-annoucing-clearly/136885/5 and https://github.com/kuzinmv/osmpie-doc/issues/9

**Comment by dabreegster** [comment]
https://community.openstreetmap.org/t/complex-ways-this-isnt-about-sidewalks/137109/19 discussion on the data model

**Comment by fhk** [comment]
just came across this today - https://ricofritzsche.me/building-a-road-network-from-osm-roads-segments-and-directed-edges/
